### ISSUE TITLE
Checkstyle Ant prints null for checkstyle version

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#6001]
https://travis-ci.org/checkstyle/checkstyle/jobs/399849113#L797
>
```
     [echo] Checkstyle started (checkstyle_checks.xml): 04/07/2018 03:42:24 AM
[checkstyle] Running Checkstyle null on 1004 files
     [echo] Checkstyle finished (checkstyle_checks.xml) : 04/07/2018 03:43:22 AM
```

`CheckstyleAntTaskTest.testExecuteLogOutput` verifies the start of the message is printed, not the version number.

**Comment by rnveach** [comment]
8.10.1 doesn't have this issue as it prints the version number correctly.
> [checkstyle] Running Checkstyle 8.10.1 on 977 files

**Comment by romani** [comment]
yes, this is a reason.

It still works fine when running by JAR. 

I launched MainTest:testVersionPrint in debug and `Main.class.getPackage().getImplementationVersion()` return `null`, test excepts null.
the same as `CheckstyleAntTask.class.getPackage().getImplementationVersion();` when running CheckstyleAntTaskTest:testExecuteLogOutput, test do not validate value of version.


content of `checkstyle-8.11-all.jar/META-INF/MANIFEST.MF`:
```
Manifest-Version: 1.0
Implementation-Title: checkstyle
Implementation-Version: 8.11
Built-By: rivanov
Specification-Title: checkstyle
Implementation-Vendor-Id: com.puppycrawl.tools
Created-By: Apache Maven 3.3.9
Build-Jdk: 1.8.0_162
Specification-Version: 8.11
Implementation-URL: http://checkstyle.sourceforge.net/
Main-Class: com.puppycrawl.tools.checkstyle.Main
```
Implementation like this exists for many years from commit  - https://github.com/checkstyle/checkstyle/commit/f85edb712767e01dafb8bf4a4a07d0d0ed3e9a38
unfortunately at that time not that much was enforced in code quality.

**Comment by romani** [comment]
@rnveach , there is no ease way to resolve this.

1)
Properties like `    <property name="checkstyle.version" value="${mvn.project.version}"/>` in our ANT config are only for resolution in configuration file https://checkstyle.org/anttask.html#Parameters see `properties`, in code [here](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java#L379).
We can probably extend scope where we can apply such variables, but it will looks like hack.

2)
```
Optional<String> versionOpt = Optional.ofNullable(
       CheckstyleAntTask.class.getPackage().getImplementationVersion());
versionOpt.orElse(System.getProperty("checkstyle.version"));
final String version = versionOpt.get();
```
there is not way to create system variable in our checkstyle ANT task to be visible in runtime to make this code work. It will require to make new property for CheckstyleAntTask like `sysproperty`, [example](https://ideoplex.com/2003/07/11/setting-java-system-properties-with-ant/), [ant spec see sysproperty](https://ant.apache.org/manual/Tasks/java.html).

3) same as "2)" but try to use environment variables.
Java can not (without hacks) to set environment variable, so it will be hard to cover by tests/pitest. So it will not work.

--------------

@rnveach , looks like "1)" and "2)" are possible way out , BUT do we really need to extend our ANT support to much ? just to make pretty output in our build log. I do not think so.

-----------

To make less scary we can also skip printing version if it is null, 
```
     [echo] Checkstyle started (checkstyle_checks.xml): 04/07/2018 01:20:55 PM
[checkstyle] Running Checkstyle  on 1004 files
     [echo] Checkstyle finished (checkstyle_checks.xml) : 04/07/2018 01:20:57 PM
```
no problems with pitest and coverage for fix:
```
$ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java b/src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java
index 9dd9aab..87e3709 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java
@@ -29,6 +29,7 @@ import java.util.Arrays;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Properties;
 import java.util.stream.Collectors;
 
@@ -335,7 +336,9 @@ public class CheckstyleAntTask extends Task {
         log("To locate the files took " + (endTime - startTime) + TIME_SUFFIX,
             Project.MSG_VERBOSE);
 
-        log("Running Checkstyle " + checkstyleVersion + " on " + files.size()
+        log("Running Checkstyle "
+                + Objects.toString(checkstyleVersion, "")
+                + " on " + files.size()
                 + " files", Project.MSG_INFO);
         log("Using configuration " + config, Project.MSG_VERBOSE);
```

@rnveach , are you ok with such update ?

**Comment by rnveach** [comment]
> no problems with pitest and coverage for fix:

There was no problems when we corrupted the version display in the first case. Pitest or our tests are incomplete for ant in some way. Something should have failed when the version was changed to `null`.

> are you ok with such update ?

If we are sure this only happens in our code and not in any other maven or ant with jar run then I am fine  when `null` is returned either dropping the version display or displaying something like `head`/`master` instead.
